From: Jan Kara <jack@suse.cz>
Subject: printk: Make printk asynchronous by default
References: bnc#744692, bnc#789311
Patch-mainline: Never, upstream decided for old behavior by default

Make printk asynchronous by default to avoid lockups during booting of large
machines. Arguably this is a better choice for SLE since the higher chances
of loosing messages on crash are not a big concern in presence of crash dumps
we use.

Signed-off-by: Jan Kara <jack@suse.cz>

---
 kernel/printk/printk.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -286,7 +286,7 @@ static char *log_buf = __log_buf;
 static u32 log_buf_len = __LOG_BUF_LEN;
 
 /* Control whether printing to console must be synchronous. */
-static bool __read_mostly printk_sync = true;
+static bool __read_mostly printk_sync = false;
 /* Printing kthread for async printk */
 static struct task_struct *printk_kthread;
 /* When `true' printing thread has messages to print */
